src/business-logic/services/inews-gateway-connection-implementation.ts

```diff
@@ -0,0 +1,123 @@
+import { WebSocket } from 'ws'
+import { Device } from '../../model/entities/device'
+import { DeviceConnection } from './interfaces/device-connection'
+import { DeviceType } from '../../model/enums/device-type'
+import { Logger } from '@tv2media/logger/*'
+
+export class INewsGatewayDeviceConnection implements DeviceConnection {
+  private static device: Device
+  private readonly INEWS_GATEWAY_HOST: string = process.env.INEWS_GATEWAY_HOST ?? 'ws://localhost:3008'
+
+  private client: WebSocket
+  private pingTimeout: NodeJS.Timeout
+  private isConnectingOrDisconnecting: boolean = false
+
+  constructor(device: Device, private readonly logger: Logger) {
+    if(INewsGatewayDeviceConnection.device?.type === DeviceType.INEWS_GATEWAY){
+      return 
+    }
```

Comment:
Since this is an implementation for handling device connection only for iNews Gateway devices, the constructor signature can be reduced from taking a generic device to taking a INewsGatewayDevice.

Reply:
Now takes the specific device type instead.